tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -210,6 +212,20 @@ build_compressed_scan_pathkeys(SortInfo *sort_info, PlannerInfo *root, List *chu
 
 			sortop =
 				get_opfamily_member(pk->pk_opfamily, var->vartype, var->vartype, pk->pk_strategy);
+			if (!get_ordering_op_properties(sortop, &opfamily, &opcintype, &strategy))
+			{
+				if (type_is_enum(var->vartype))
+				{
+					sortop = get_opfamily_member(pk->pk_opfamily,
+												 ANYENUMOID,
+												 ANYENUMOID,
+												 pk->pk_strategy);
+				}
+				else
+				{
+					elog(ERROR, "Invalid segmentby column");
```

Comment:
This does not conform to pg error message style guide: https://www.postgresql.org/docs/current/error-style-guide.html
Should probably be more along the lines of `sort operator lookup failed for column "%s"`

Reply:
Fixed as per your suggestion.